pkg/apis/externaldns/types.go

```diff
@@ -61,28 +58,34 @@ func NewConfig() *Config {
-	flags.StringVar(&cfg.Compatibility, "compatibility", "", "enable to process annotation semantics from legacy implementations: <mate|molecule>")
-	flags.StringVar(&cfg.MetricsAddress, "metrics-address", defaultMetricsAddress, "address to expose metrics on")
-	flags.StringVar(&cfg.LogFormat, "log-format", defaultLogFormat, "log format output: <text|json>")
-	flags.DurationVar(&cfg.Interval, "interval", time.Minute, "interval between synchronizations")
-	flags.BoolVar(&cfg.Once, "once", false, "run once and exit")
-	flags.BoolVar(&cfg.DryRun, "dry-run", true, "run without updating DNS provider")
-	flags.BoolVar(&cfg.Debug, "debug", false, "debug mode")
-	flags.BoolVar(&cfg.Version, "version", false, "display the version")
-	flags.StringVar(&cfg.Registry, "registry", "noop", "type of registry for ownership: <noop|txt>")
-	flags.StringVar(&cfg.RecordOwnerID, "record-owner-id", "", "id for keeping track of the managed records")
-	flags.StringVar(&cfg.TXTPrefix, "txt-prefix", "", `prefix assigned to DNS name of the associated TXT record; e.g. for --txt-prefix=abc_ [CNAME example.org] <-> [TXT abc_example.org]`)
-	flags.StringVar(&cfg.FqdnTemplate, "fqdn-template", "", `fallback template to generate DNS name if annotation is missing; 
-		e.g. --fqdn-template={{.Name}}-{{.Namespace}}.example.org will use service/ingress name/namespace`)
-	return flags.Parse(args)
+	app := kingpin.New("external-dns", "ExternalDNS synchronizes exposed Kubernetes Services and Ingresses with DNS providers.")
+	app.Version(version)
+	app.DefaultEnvars()
+
+	app.Flag("in-cluster", "Retrieve target cluster configuration from the environment").BoolVar(&cfg.InCluster)
+	app.Flag("kubeconfig", "Retrieve target cluster configuration from a Kubernetes configuration file").StringVar(&cfg.KubeConfig)
+	app.Flag("namespace", "Limit sources of endpoints to a specific namespace; defaults to all").StringVar(&cfg.Namespace)
+	app.Flag("domain", "Limit possible target zones by a domain suffix").StringVar(&cfg.Domain)
+	app.Flag("source", "The resource types that are queried for endpoints").StringsVar(&cfg.Sources)
+	app.Flag("provider", "The DNS provider where the DNS records will be created").StringVar(&cfg.Provider)
+	app.Flag("google-project", "When using the Google provider, specify the Google project").StringVar(&cfg.GoogleProject)
+	app.Flag("policy", "Modify how DNS records are sychronized between sources and providers; possible choices: <sync|upsert-only> default: sync").Default("sync").StringVar(&cfg.Policy)
+	app.Flag("compatibility", "Process annotation semantics from legacy implementations; possible choices: <mate|molecule>, default: none").StringVar(&cfg.Compatibility)
+	app.Flag("metrics-address", "Specify were to serve the metrics and health check endpoint").Default(defaultMetricsAddress).StringVar(&cfg.MetricsAddress)
+	app.Flag("log-format", "The format in which log messages are printed").Default(defaultLogFormat).StringVar(&cfg.LogFormat)
```

Comment:
Do you also want to say what are the possible formats? Like this I wouldn't know.